### PR TITLE
ci: add jacoco code coverage to the ci

### PR DESCRIPTION
**Pull request by Wylwi**
Sonar is working on top of the Jacoco report, but doesn't compute it the same way. This upload the report to github actions. 

This PR also adds a small jacoco parser, to show more detailed description in the PR, without having to go all the way to the zip file.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -127,7 +127,30 @@ jobs:
       # Upload the various reports to sonar
       - name: Upload report to SonarCloud
         env:
-          GITHUB_TOKEN: ${{ secrets.TOKEN_GITHUB }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (rBergCode) [inline .github/workflows/ci.yml:130]: Question: why was this changed ?

File: .github/workflows/ci.yml
Comment (rBergCode) [inline .github/workflows/ci.yml:1]: Praise: nice ! The message looks good!

**Review by rBergCode — COMMENTED** [review]
I have one question and you'll be good to go! Nice job!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Wylwi) [reply .github/workflows/ci.yml:130]: I initially created a TOKEN_GITHUB secret with a handmade token, but GitHub, when launching the CI, gives a GITHUB_TOKEN by default, which is a secret specifically created for this specific CI run. This is just a security improvement

**Review by rBergCode — APPROVED** [review]
Thanks!
